burn-import/src/onnx/convert.rs

```diff
@@ -0,0 +1,484 @@
+use std::{
+    collections::{HashMap, HashSet},
+    fs::File,
+    path::Path,
+    str::{from_utf8, FromStr},
+};
+
+use super::coalesce::coalesce;
+use super::ir::{
+    ArgType, Argument, AttributeValue, Attributes, ElementType, Graph, Node, NodeType, Tensor,
+    TensorData,
+};
+use super::protos::{
+    attribute_proto::AttributeType, tensor_proto::DataType, tensor_shape_proto::dimension::Value,
+    type_proto, AttributeProto, ModelProto, NodeProto, TensorProto, TensorShapeProto,
+    ValueInfoProto,
+};
+use super::shape_inference::shape_inference;
+
+use protobuf::{Enum, Message};
+use topological_sort::TopologicalSort;
+
+const STATEFUL_NODE_TYPES: [NodeType; 4] = [
+    NodeType::Conv,
+    NodeType::BatchNormalization,
+    NodeType::Dropout,
+    NodeType::Linear,
+];
+
+#[derive(Debug)]
+/// Error type for parsing ONNX model
+pub enum ParseError {
+    VariantNotFound,
+}
+
+/// Open an onnx file and convert it to a Graph (intermediate representation)
+pub fn parse_onnx(onnx_path: &Path) -> Graph {
+    let mut file = File::open(onnx_path).expect("Unable to open file");
+    let onnx_model: ModelProto =
+        Message::parse_from_reader(&mut file).expect("Unable to parse ONNX file");
+
+    // Convert the nodes
+    let mut nodes: Vec<Node> = vec![];
+    for onnx_node in onnx_model.graph.node.iter() {
+        nodes.push(convert_node_proto(onnx_node));
+    }
```

Comment:
It's more useful doing in Vec because the nodes need to be sorted topologically (keyed by node inputs/outputs). There is currently no use case to loop up by node's name, only by inputs and outputs. 

Initially I was using map but it added up being more difficult to use.